core/src/main/scala/org/apache/spark/scheduler/TaskResultGetter.scala

```diff
@@ -81,8 +81,9 @@ private[spark] class TaskResultGetter(sparkEnv: SparkEnv, scheduler: TaskSchedul
                   taskSetManager, tid, TaskState.FINISHED, TaskResultLost)
                 return
               }
+              // TODO(josh): assumption that there is only one chunk here is a hack
               val deserializedResult = serializer.get().deserialize[DirectTaskResult[_]](
-                serializedTaskResult.get)
+                serializedTaskResult.get.getChunks().head)
```

Comment:
This is a messy temporary hack caused by some interface mismatching; I'll see about fixing this so that we don't make any assumptions about there only being one chunk here.